components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+                                if (inputEl.type !== 'checkbox') {
+                                    setTimeout(() => this.validateField(item.id, item.value), 250);
+                                }
+                            };
+                        }
+
+                        if (!isHidden) {
+
+                            if (this.mode === 'override') {
+                                // Wrap the inputEl in an override.
+                                const overrideEl = createElementLegacy(null, 'ca-override', { name: key });
+                                overrideEl.property = item;
+                                overrideEl.appendChild(inputEl);
+                                inputEl = overrideEl;
+                            }
+
+                            const itemDesc = item.description || '';
+                            const itemDescHasHtml = /<[^>]*>/gi.test(itemDesc || '');
+
+                            // if it contains html
+                            if (itemDescHasHtml) {
+
+                                // inject as safe HTML
+                                const descEl = createElementLegacy(lbl, 'i', { class: 'ca-field-desc' });
+
+                                htmlToDom(itemDesc, descEl);
+                                // or has length over placeholder max length characters
+                            } else if (itemDesc.length >= this.placeholderMaxLength) {
+
+                                // inject as text
+                                createElementLegacy(lbl, 'i', { class: 'ca-field-desc' }, itemDesc);
+                            }
+
+                            // add input to form label
+                            lbl.appendChild(inputEl);
+
+                            // add row to form
+                            parentEl.appendChild(lbl);
+                        } else {
+                            parentEl.appendChild(inputEl);
+                        }
+
+                        if (this.columns > 1 && itemsPerCol >= maxItemsPerCol && itemIndex < len) {
+
+                            // add a new column
+                            group = createElementLegacy(this.form, 'fieldset');
+                            parentEl = createElementLegacy(group, 'div', { class: 'group' });
+
+                            // increase the column count
+                            columnCount++;
+
+                            // adjust the maxItemsPerCol on the basis of how many items are left to render across the remaining columns
+                            maxItemsPerCol = (len - itemIndex) / (this.columns - columnCount);
+
+                            // reset counter
+                            itemsPerCol = 0;
+                        }
+                    } else {
+                        // eslint-disable-next-line no-console
+                        console.log(`In ca-form.render(): Failed to convert schema item to html element (key: ${key})`);
```

Comment:
I'm going to use the trace module for now instead of console.log. We should have a proper discussion around error handling though.